### PR TITLE
Allow empty configuration for SLM policies

### PR DESCRIPTION
**Pull request by dakrone**
When putting or updating a snapshot lifecycle policy it was not possible
to elide the `config` map. This commit makes the configuration optional,
the same way that it is when taking a snapshot.

Relates to #38461

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/snapshotlifecycle/SnapshotLifecyclePolicy.java [hunk 1]
```diff
@@ -61,11 +59,11 @@
...
-        PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, c) -> p.map(), CONFIG);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> p.map(), CONFIG);
     }
 
     public SnapshotLifecyclePolicy(final String id, final String name, final String schedule,
-                                   final String repository, Map<String, Object> configuration) {
+                                   final String repository, @Nullable Map<String, Object> configuration) {
         this.id = Objects.requireNonNull(id);
         this.name = name;
         this.schedule = schedule;
```
Comment (jakelandis) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/snapshotlifecycle/SnapshotLifecyclePolicy.java]: nit: it is unclear if `name`, `schedule`, and `repository` are `@Nullable` If so, can you add the annontation... if not can you Objects.requireNonNull()` ?

**Review by jakelandis — APPROVED** [review]
LGTM (one minor nitpick)

**Review by jakelandis — APPROVED** [review]
LGTM (one minor nitpick)

**Comment by dakrone** [comment]
@elasticmachine update branch

**Pull request merged**